chia/pools/pool_puzzles.py

```diff
@@ -351,14 +356,21 @@ def solution_to_extra_data(full_spend: CoinSolution) -> Optional[PoolState]:
 
     # Not launcher spend
     inner_solution: Program = full_solution.rest().rest().first()
-    inner_spend_type: int = inner_solution.first().as_int()
-
-    if inner_spend_type == 0:
-        # Absorb
-        return None
 
     # Spend which is not absorb, and is not the launcher
-    extra_data = inner_solution.rest().rest().rest().first().as_atom()
+    num_args = len(inner_solution.as_python())
+    assert num_args in (2, 3)
+
+    if num_args == 2:
+        # pool member
+        if inner_solution.rest().first().as_int() == 1:
+            return None
+        extra_data = inner_solution.first().first().rest().as_atom()
```

Comment:
This presumes we only have one key, which may not be true in future if we add more fields. You ought to iterate over the key-value pairs, looking for `p`.